os/arch/arm/src/s5j/s5j_i2c.c

```diff
@@ -765,7 +765,7 @@ int s5j_i2c_transfer(struct i2c_dev_s *dev, struct i2c_msg_s *msgv, int msgc)
 			ret = readbytes(priv, pmsg);
 			if (ret < pmsg->length) {
 				if (ret >= 0) {
-					return -EIO;
+					ret = -EIO;
```

Comment:
here is strange.
Assume that pmsg->length is 2, and ret is 1
in this case, it goes to fail and return ok but priv->ptr is null
is it right operation?
I guess below is right
if (pmsg->flags & I2C_M_READ) {
       .
       .
        if (ret < pmsg->length) {
             goto fail;
        }
}  else {
         .
         if(ret < pmsg->length) {
               goto fail;
          }
}

fail : 
     if (ret > =0) {
             ret = -EI);
     }
     return ret;

If I am wrong, ignore this comment.

Reply:
Assume that pmsg->length is 2, and ret is 1.
ret >= 0 is true, so ret will be -EIO and s5j_i2c_transfer returns -EIO, not ok.